src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -182,6 +199,33 @@ public DeploymentResult call() throws InterruptedException {
         }
     }
 
+    private boolean isStaleDeployment(DeploymentDocument deploymentDocument) {
+        // Check if group deployment
+        boolean isGroupDeployment = Deployment.DeploymentType.IOT_JOBS.equals(deployment.getDeploymentType())
+                                        && deploymentDocument.getGroupName() != null;
+
+        // if not a group deployment, then not stale
+        if (!isGroupDeployment) {
+            return false;
+        }
+
+        // Get timestamp for the root target group
+        Topics lastDeployment = deploymentServiceConfig
+                .lookupTopics(DeploymentService.GROUP_TO_LAST_DEPLOYMENT_TOPICS)
+                .lookupTopics(deploymentDocument.getGroupName());
+
+        long timestamp = Coerce.toLong(lastDeployment.find(GROUP_TO_LAST_DEPLOYMENT_TIMESTAMP_KEY));
+
+        // if don't have last deployment detail, then its a new deployment
+        if (timestamp == 0 || deploymentDocument.getTimestamp() == null) {
+            return false;
+        }
+
+        // return stale only if new deployment's timestamp is older
+        // than last deployment
+        return deploymentDocument.getTimestamp().compareTo(timestamp) < 0;
```

Comment:
since timestamp is a long, just using < seems more clear to me than using compareTo.